Lesson34/Merve/Lesson33/props-drilling-app/src/modules/user/UserDetailsForm/index.jsx

```diff
@@ -0,0 +1,33 @@
+import { useState, useContext } from "react";
+import { UserDispatchContext } from "../UserProvider";
+import './styles.css';
+
+export const UserDetailsForm = ({ username, onNameChange }) => {
+  const [newName, setNewName] = useState(username);
+  const setUser = useContext(UserDispatchContext);
+
+  const handleChange = (e) => {
+    setNewName(e.target.value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setUser((prevUser) => ({
+      ...prevUser,
+      name: newName
+    }));
+    onNameChange(newName);
+  };
+
+  return (
+    <form className="user-details-form" onSubmit={handleSubmit}>
+      <input
+        type="text"
+        value={newName}
+        onChange={handleChange}
+        placeholder="Change username"
```

Comment:
For placeholder value, we want to show a user an example, not to provide instructions so using 'Change username' is not correct here. Better would be something like 'johndoe13' assuming that's a format that we want to user for usernames in our application.